spec/addressable/uri_spec.rb

```diff
@@ -4262,6 +4262,36 @@ def to_s
   end
 end
 
+describe Addressable::URI, "when parsed from 'https://example.com/?q=a+b'" do
+  before do
+    @uri = Addressable::URI.parse("https://example.com/?q=a+b")
+  end
+
+  it "should have query_values of {'q' => 'a b'}" do
+    expect(@uri.query_values).to eq({'q' => 'a b'})
+  end
+end
+
+describe Addressable::URI, "when parsed from 'example.com?q=a+b'" do
+  before do
+    @uri = Addressable::URI.parse("example.com?q=a+b")
+  end
+
+  it "should have query_values of {'q' => 'a b'}" do
+    expect(@uri.query_values).to eq({'q' => 'a b'})
+  end
+end
+
+describe Addressable::URI, "when parsed from 'mailto:?q=a+b'" do
+  before do
+    @uri = Addressable::URI.parse("mailto:?q=a+b")
+  end
+
+  it "should have query_values of {'q' => 'a+b'}" do
+    expect(@uri.query_values).to eq({'q' => 'a+b'})
```

Comment:
Was largely copied from the spec above. Happy to do either of:

1. Fix just these three violations, leaving the other specs inconsistent with these ones
2. Fix these three violations and all similar violations nearby
3. Leave it alone since it is consistent with the existing specs

Reply:
I'd go for 1., and leave the inconsistencies for a future PR that addresses them explicitly (just my preference, I would be fine with any of the 3)